test/one/profiler/test/Runner.java

```diff
@@ -210,24 +204,59 @@ private static void printSummary(EnumMap<TestStatus, Integer> statusCounts, List
         System.out.println("TOTAL: " + testCount);
     }
 
+    private static RunnerDeclaration parseRunnerDeclaration(String[] args) {
+        // All available directories.
+        List<String> testDirectories = new ArrayList<>();
+
+        // If directories are specified, they are considered an exact match.
+        List<String> includeDirs = new ArrayList<>();
+
+        // Glob filters matching "ClassName.methodName".
+        List<String> testFilters = new ArrayList<>();
+
+        for (int i = 0; i < args.length; i++) {
+            String arg = args[i];
+            File f = new File("test/test/" + arg);
+            if (!f.exists() || !f.isDirectory()) {
```

Comment:
RunnerDeclaration already gets a list of all directories elsewhere. Let's use list (or better, set) here.

Reply:
We need to check directory existence to distinguish between `testFilters` and `includeDirs`.

Reply:
You get complete list of test directories when Runner starts, no need to do filesystem checks again. `includeDirs` will go away as per my other comment.
So, the algorithm I imagine works as follows:
1. Obtain a set of test directories by listing `test/test/`.
2. Parse test specifiers passed in command line arguments, converting each one to `Pattern`.
3. Similarly, parse `skip` definition to get a list of Patterns to be skipped.
4. Use reflection to find all runnable tests from directories obtained at (1).
5. Go through a sorted list of runnable tests and execute them if the name matches any pattern from (2) and does not match any pattern from (3).

Does this make sense?